src/core/tests/library/LegacyEnvLayerExtensions.py

```diff
@@ -803,9 +803,15 @@ def run_command_output(self, cmd, no_output=False, chk_err=True):
                         code = 100
                         output = "E: dpkg was interrupted, you must manually run 'sudo dpkg --configure -a' to correct the problem."
 
-            if sys.version_info.major == 2:
+            major_version = None
+            if hasattr(sys.version_info, 'major'):
+                major_version = sys.version_info.major
+            else:
+                major_version = sys.version_info[0]
```

Comment:
Even though it's only used once in this area, encapsulate into a different function in the same class for code readability.